src/fillTank.js

```diff
@@ -14,17 +14,38 @@
  * @param {number} amount
  */
 function fillTank(customer, fuelPrice, amount = Infinity) {
-  const { vehicle } = customer;
+  const { vehicle, money } = customer;
+
+  if (typeof customer !== 'object'
+    || customer === null || Array.isArray(customer)) {
+    throw new Error('customer should be only an object');
+  }
+
+  if ((!money && money !== 0)
+    || (!vehicle.fuelRemains && vehicle.fuelRemains !== 0)
+    || (!vehicle.maxTankCapacity || vehicle.maxTankCapacity <= 0)
+    || !vehicle) {
```

Comment:
The condition checks for `money`, `vehicle.fuelRemains`, and `vehicle.maxTankCapacity` are mostly correct. However, the check `(!money && money !== 0)` can be simplified to `money == null` to check for both `null` and `undefined`. Similarly, `(!vehicle.fuelRemains && vehicle.fuelRemains !== 0)` can be simplified to `vehicle.fuelRemains == null`.

Reply:
done